lib/IDE/CodeCompletion.cpp

```diff
@@ -2973,6 +3011,47 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
     if (AFT && !IsImplicitlyCurriedInstanceMethod)
       trivialTrailingClosure = hasTrivialTrailingClosure(FD, AFT);
 
+    Optional<CodeCompletionResult::NotRecommendedReason> NotRecommended;
+    bool implictlyAsync = false;
+    auto isolation = getActorIsolation(const_cast<FuncDecl *>(FD));
+    switch (isolation.getKind()) {
```

Comment:
Can this be factored out so that we don't have so much duplication between the VarDecl and FuncDecl cases?  Having it in one place would help avoid forgetting to update them all if anything changes as well.

Or maybe there is a way we could share more of this implementation with the typechecker?  I see `diagnoseNonConcurrentTypesInReference` has similar logic around checking the specific decl, although the ActorIsolation dispatch is at a higher level I guess.